homeassistant/components/dsmr/sensor.py

```diff
@@ -405,6 +394,35 @@ class DSMRSensorEntityDescription(
 )
 
 
+def add_gas_sensor_5B(
+    telegram: dict[str, DSMRObject]
+) -> tuple[DSMRSensorEntityDescription]:
+    """Return correct entity for 5B Gas meter."""
+    ref = None
+    if obis_references.BELGIUM_MBUS4_METER_READING2 in telegram:
+        ref = obis_references.BELGIUM_MBUS4_METER_READING2
+    if obis_references.BELGIUM_MBUS3_METER_READING2 in telegram:
+        ref = obis_references.BELGIUM_MBUS3_METER_READING2
+    if obis_references.BELGIUM_MBUS2_METER_READING2 in telegram:
+        ref = obis_references.BELGIUM_MBUS2_METER_READING2
+    if obis_references.BELGIUM_MBUS1_METER_READING2 in telegram:
+        ref = obis_references.BELGIUM_MBUS1_METER_READING2
+    if ref is None:
```

Comment:
```suggestion
    if obis_references.BELGIUM_MBUS1_METER_READING2 in telegram:
        ref = obis_references.BELGIUM_MBUS1_METER_READING2
    elif obis_references.BELGIUM_MBUS2_METER_READING2 in telegram:
        ref = obis_references.BELGIUM_MBUS2_METER_READING2
    elif obis_references.BELGIUM_MBU3_METER_READING2 in telegram:
        ref = obis_references.BELGIUM_MBUS3_METER_READING2
    elif obis_references.BELGIUM_MBUS4_METER_READING2 in telegram:
        ref = obis_references.BELGIUM_MBUS4_METER_READING2
    elif ref is None:
```

Avoids all conditons to be checked, as you can have only one. (also changed the order)